editoast/src/views/timetable/track_occupancy.rs

```diff
@@ -175,69 +96,93 @@ pub fn find_track_occupancy_for_operational_point(
     pathfinding: &PathfindingResult,
     train_schedule: &schemas::TrainSchedule,
 ) -> Vec<TrackOccupancy> {
-    // Extract pathfinding data
-    let (track_ranges, path_item_positions) = match pathfinding {
-        PathfindingResult::Success(pathfinding_result_success) => {
-            assert_eq!(
-                pathfinding_result_success.path_item_positions.len(),
-                train_schedule.path.len()
-            );
-            (
-                &pathfinding_result_success.path.track_section_ranges,
-                &pathfinding_result_success.path_item_positions,
-            )
-        }
-        PathfindingResult::Failure(_) => {
-            // The timetable can have a pathfinding failure for a train schedule
-            // because the train schedule is not valid.
-            // We skip it because we don't want to compute the track occupancy for an invalid train schedule.
-            // It's not a problem because we are just looking for the track occupancy of the operational point for the valid trains.
-            tracing::info!(train_schedule.train_name, "pathfinding failed");
-            return vec![];
+    let (is_valid, final_output, pathfinding_success) = match (simulation, pathfinding) {
+        (
+            simulation::Response::Success(SimulationResponseSuccess { final_output, .. }),
+            PathfindingResult::Success(pathfinding_success),
+        ) => (true, Some(final_output), Some(pathfinding_success)),
+        (_, PathfindingResult::Success(pathfinding_success)) => {
+            (false, None, Some(pathfinding_success))
         }
+        _ => (false, None, None),
```

Comment:
1. `is_valid` is not useful (can be deduced from other variable with `final_output.is_some()`
2. We can extract PF and Sim separately


```suggestion
    let final_output = match simulation {
        simulation::Response::Success(SimulationResponseSuccess { final_output, .. }) => {
            Some(final_output)
        }
        _ => None,
    };

    let pathfinding_success = match pathfinding {
        PathfindingResult::Success(pathfinding_success) => Some(pathfinding_success),
        _ => None,
    };
```

Reply:
Done